cfme/tests/automate/custom_button/test_generic_objects.py

```diff
@@ -157,3 +170,65 @@ def test_custom_button_automate(appliance, request, submit, setup_obj, button_gr
             assert False, "Expected {} requests not found in automation log".format(
                 str(expected_count)
             )
+
+
+@pytest.mark.ignore_stream("5.9")
+def test_custom_button_dialog(appliance, dialog, request, setup_obj, button_group):
+    """ Test custom button with dialog and InspectMe method
+
+    Prerequisites:
+        * Appliance
+        * Simple TextInput service dialog
+
+    Steps:
+        * Create custom button group with the Object type
+        * Create a custom button with service dialog
+        * Navigate to object Details page
+        * Check for button group and button
+        * Select/execute button from group dropdown for selected entities
+        * Fill dialog and submit
+        * Check for the proper flash message related to button execution
+        * Check request in automation log
+    """
+
+    group, obj_type = button_group
+
+    # Note: No need to set display_for dialog only work with Single entity
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        dialog=dialog,
+        system="Request",
+        request="InspectMe",
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+    custom_button_group.item_select(button.text)
+
+    dialog_view = view.browser.create_view(TextInputDialogView)
+    dialog_view.wait_displayed()
+    assert dialog_view.service_name.fill("Custom Button Execute")
+
+    # Clear the automation log
+    assert appliance.ssh_client.run_command('echo -n "" > /var/www/miq/vmdb/log/automation.log')
+
+    # Submit order
+    dialog_view.submit.click()
+
+    view.wait_displayed("60s")
```

Comment:
Question : Is this required ? 
The flash message appears after a delay ?

Reply:
Sometimes it took some delay; you can say due to network... I found this some time while local test so I added... I know its much `60s` but it's not a problem as `is_displayed` return `True` loop breaks.